### PR TITLE
feat: add version header

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This adds a version header to the installation of EDM4eic. This allows us to make downstream code behave differently based on the version. 

Note: with `.gitattributes` (in particular `export-subst`) we could automate the filling of this version information with the most recent tag and such. Maybe something to look into.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,10 +2,15 @@
 # Copyright (C) 2022 Whitney Armstrong, Sylvester Joosten, Wouter Deconinck
 
 cmake_minimum_required(VERSION 3.12)
+
 project(EDM4EIC
-  VERSION 1.1.0
   LANGUAGES CXX)
 
+SET( ${PROJECT_NAME}_VERSION_MAJOR 2 )
+SET( ${PROJECT_NAME}_VERSION_MINOR 0 )
```
Comment (veprbl) [inline CMakeLists.txt:10]: Safe to put this to the next release version
```suggestion
SET( ${PROJECT_NAME}_VERSION_MINOR 1 )
```

**Review by veprbl — DISMISSED** [review]

File: EDM4eicVersion.h.in [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+/// Get the minor version from a preprocessor defined version
+#define EDM4EIC_MINOR_VERSION(v) (((v) & (-1UL >> 32)) >> 16)
+/// Get the patch version from a preprocessor defined version
+#define EDM4EIC_PATCH_VERSION(v) ((v) & (-1UL >> 48))
+
+// Some helper constants that are populated by the cmake configure step
+#define EDM4EIC_VERSION_MAJOR @EDM4EIC_VERSION_MAJOR@
+#define EDM4EIC_VERSION_MINOR @EDM4EIC_VERSION_MINOR@
+#define EDM4EIC_VERSION_PATCH @EDM4EIC_VERSION_PATCH@
+#define EDM4EIC_VERSION EDM4EIC_VERSION(EDM4EIC_VERSION_MAJOR, EDM4EIC_VERSION_MINOR, EDM4EIC_VERSION_PATCH)
```
Comment (veprbl) [inline EDM4eicVersion.h.in:28]: Double definition?

Comment (wdconinc) [reply EDM4eicVersion.h.in:28]:
```suggestion
#define edm4hep_VERSION EDM4EIC_VERSION(EDM4EIC_VERSION_MAJOR, EDM4EIC_VERSION_MINOR, EDM4EIC_VERSION_PATCH)
```

Comment (wdconinc) [reply EDM4eicVersion.h.in:28]: https://github.com/key4hep/EDM4hep/blob/main/EDM4hepVersion.h.in#L27C9-L27C24

**Pull request merged**